scripts/cat/names.py

```diff
@@ -173,12 +175,16 @@ def give_suffix(self, pelt, biome, tortiepattern):
                 self.suffix = random.choice(self.names_dict["normal_suffixes"])
 
     def __repr__(self):
+        # Handles predefined suffixes (such as newborns being kit), then suffixes based on ages (fixes #2004, just trust me)
         if self.status in self.names_dict["special_suffixes"] and not self.specsuffix_hidden:
             return self.prefix + self.names_dict["special_suffixes"][self.status]
-        else:
-            if game.config['fun']['april_fools']:
-                return self.prefix + 'egg'
-            return self.prefix + self.suffix
+        if self.status in ['loner'] and self.moons is not None and self.moons < 12:
```

Comment:
Moons <12 doesn't necessarily line up with being an apprentice; a lot of cases here where that wouldn't line up. Such as when players have exp based graduation turned on and the cats graduate based on exp. If you really wanted to, you could check for exp based graduation and check the cat's exp to see if they are really meant to be an apprentice. But that leaves a bit of code redundancy - if anything changes with the app/exp code it could mess up here. Maybe the apprentice graduation already has a function you could borrow for that. 
Though that might not fix things in the case of, like, Ravenpaw, where he kept his apprentice name his whole life after leaving. Not sure there's a good way to approach this.

Reply:
This will only be used for loners outside the clan though, such as in one of the outcomes for `gen_bord_welcomingsparrowgift1` where you meet a loner kitten. It would appear as if loners don't gain any experience, and if they join the clan then they have their loner status stripped.
I personally don't see another way to fix this outside of potentially adding an `is_loner` flag to each cat, but that sounds unneeded.